src/core/public/injected_metadata/injected_metadata_service.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { deepFreeze } from './deep_freeze';
+
+export interface InjectedMetadataParams {
+  injectedMetadata: {
+    legacyMetadata: {
+      [key: string]: any;
+    };
+  };
+}
+
+export class InjectedMetadataService {
+  constructor(private params: InjectedMetadataParams) {}
```

Comment:
__nit:__ `private readonly`

Reply:
The main benefit I see is that when I read constructor or class definition I know for sure this class member won't change and don't have to scan class to see where it's mutated. But yeah, I don't have strong opinion, I just like "immutable by default" rule and make immutable everything I can.